sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -241,9 +245,79 @@ protected internal string GetSessionFilePath()
 
             string fileName = $"{name}{version}{async}.json";
 
-            return Path.Combine(
+            var repoRoot = GetRepoRoot();
+
+            // this needs to be updated to purely relative to repo root
+            var result = Path.Combine(
                 GetSessionFileDirectory(),
                 fileName);
+
+            return Regex.Replace(result.Replace(repoRoot, String.Empty), @"^[\\/]*", string.Empty);
+        }
+
+        private class DirectoryEvaluation
+        {
+            public bool IsRoot;
+            public bool IsGitRoot;
+            public bool AssetsJsonPresent;
+        }
+
+        private DirectoryEvaluation EvaluateDirectory(string directoryPath)
+        {
+            var assetsJsonLocation = Path.Combine(directoryPath, AssetsJson);
+            var gitLocation = Path.Combine(directoryPath, ".git");
+
+            return new DirectoryEvaluation()
+            {
+                AssetsJsonPresent = File.Exists(assetsJsonLocation),
+                IsGitRoot = Directory.Exists(gitLocation),
+                IsRoot = new DirectoryInfo(directoryPath).Parent == null
+            };
+        }
+
+        public string GetRepoRoot()
```

Comment:
We can use TestEnvironment.RepoRoot instead of this method.

Reply:
🤦 

will do 👍